clients/java/src/main/java/io/camunda/zeebe/client/impl/oauth/OAuthCredentialsCache.java

```diff
@@ -86,12 +88,54 @@ private void ensureCacheFileExists() throws IOException {
     if (cacheFile.exists()) {
       return;
     }
-
-    Files.createDirectories(cacheFile.getParentFile().toPath());
+    final File parentDirectory = cacheFile.getParentFile();
+    if (parentDirectory.exists()) {
+      if (!parentDirectory.isDirectory()) {
+        if (Files.isSymbolicLink(parentDirectory.toPath())) {
+          requireSymbolicLinkPointsToDirectory(parentDirectory);
+        } else {
+          throw new IOException(
+              "Expected "
+                  + parentDirectory.getAbsolutePath()
+                  + " to be a directory, but it was a regular file.");
+        }
+      }
+    } else {
+      if (Files.isSymbolicLink(parentDirectory.toPath())) {
+        requireSymbolicLinkPointsToDirectory(parentDirectory);
```

Comment:
Yes. This sounds crazy, but this can happen. Checkout the tests. If the path is a symbolic link, then the "exists" check is not testing if the symbolic link exists, but it resolves the symbolic link and checks if the target exists.

In short, if the symbolic link is broken (points to a directory that doesn't exist) then, it makes it is still a file and creating of a directory with this name makes no sense. Please check my unit tests for this.

I also attached you a coverage report which shows that the code is actually covered by the test (so it is not dead).

![image](https://user-images.githubusercontent.com/673128/195197656-f497b1f6-de0d-4005-a671-946de4899819.png)

Reply:
👍 Awesome, thanks for that explanation! Since the case is covered, we can also do without a comment explaining how this works. If someone tries to remove it, the test will fail.